src/libraries/System.Private.CoreLib/src/System/Guid.cs

```
@@ -261,6 +272,54 @@ public Guid(string g)
+        /// <remarks>
+        ///     <para>This corresponds to the most significant 4 bits of the 6th byte: 00000000-0000-F000-0000-000000000000.</para>
+        ///     <para>See RFC 9562 for more information on how to interpret this value.</para>
+        /// </remarks>
+        public int Version => _c >>> 12;
+
+        /// <summary>Creates a new <see cref="Guid" /> according to RFC 9562, following the Version 7 format.</summary>
+        /// <returns>A new <see cref="Guid" /> according to RFC 9562, following the Version 7 format.</returns>
+        /// <remarks>
+        ///     <para>This uses <see cref="DateTimeOffset.UtcNow" /> to determine the Unix Epoch timestamp source.</para>
+        ///     <para>This seeds the rand_a and rand_b sub-fields with random data.</para>
+        /// </remarks>
+        public static Guid CreateVersion7() => CreateVersion7(DateTimeOffset.UtcNow);
+
+        /// <summary>Creates a new <see cref="Guid" /> according to RFC 9562, following the Version 7 format.</summary>
+        /// <param name="timestamp">The date time offset used to determine the Unix Epoch timestamp</param>
+        /// <returns>A new <see cref="Guid" /> according to RFC 9562, following the Version 7 format.</returns>
+        /// <remarks>
+        ///     <para>This seeds the rand_a and rand_b sub-fields with random data.</para>
+        /// </remarks>
+        public static Guid CreateVersion7(DateTimeOffset timestamp)
+        {
+            // This isn't the most optimal way, but we don't have an easy way to get
+            // secure random bytes in corelib without doing this, and its not terrible
+            Guid result = NewGuid();
+
+            // 2^48 is roughly 8925.5 years, which from the Unix Epoch means we won't
+            // overflow until around July of 10,895. So there isn't much need to handle it
+            // particularly given that DateTimeOffset.MaxValue is December 31, 9999
```

What would need to be handied then if a larger value can't be passed in? Trying to understand the comment.

---

It's a comment that's explicitly covering why only taking 48-bits out of a `64-bit integer` is okay, that we won't end up in a scenario where we're silently dropping information.